src/bun.js/ipc.zig

```diff
@@ -197,12 +255,35 @@ const json = struct {
 
         const slice = str.slice();
 
-        try writer.ensureUnusedCapacity(slice.len + 1);
+        try writer.ensureUnusedCapacity(1 + slice.len + 1);
+
+        writer.writeAssumeCapacity(&.{1});
+        writer.writeAssumeCapacity(slice);
+        writer.writeAssumeCapacity("\n");
+
+        return 1 + slice.len + 1;
+    }
+
+    pub fn serializeInternal(_: *IPCData, writer: anytype, global: *JSC.JSGlobalObject, value: JSValue) !usize {
+        var out: bun.String = undefined;
+        value.jsonStringify(global, 0, &out);
+        defer out.deref();
+
+        if (out.tag == .Dead) return IPCSerializationError.SerializationFailed;
+
+        // TODO: it would be cool to have a 'toUTF8Into' which can write directly into 'ipc_data.outgoing.list'
+        const str = out.toUTF8(bun.default_allocator);
+        defer str.deinit();
+
+        const slice = str.slice();
+
+        try writer.ensureUnusedCapacity(1 + slice.len + 1);
 
+        writer.writeAssumeCapacity(&.{2});
```

Comment:
What is 2?

Reply:
a header for the serialized messages. 1 is regular json, 2 is internal